### DISCUSSION TITLE
Can't load under nixos?

### DISCUSSION THREAD

**Discussion by tmpm697** [olimorris/persisted.nvim#87] [General]
I can't seem to make this plugin to work with neovim under nixos. Always got this error when load plugin:
```
building '/nix/store/dj9azask9bdnp3lzni7qjj68aj5g2bjg-vimplugin-persisted-.drv'...
error: builder for '/nix/store/dj9azask9bdnp3lzni7qjj68aj5g2bjg-vimplugin-persisted-.drv' failed with exit code 2;
       last 10 log lines:
       > source root is persisted.nvim
       > patching sources
       > updateAutotoolsGnuConfigScriptsPhase
       > configuring
       > no configure script, doing nothing
       > building
       > build flags: SHELL=/nix/store/ir0j7zqlw9dc49grmwplppc7gh0s40yf-bash-5.2-p15/bin/bash
       > stylua lua/ tests/ -f ./stylua.toml
       > /nix/store/ir0j7zqlw9dc49grmwplppc7gh0s40yf-bash-5.2-p15/bin/bash: line 1: stylua: command not found
       > make: *** [Makefile:31: format] Error 127
       For full logs, run 'nix log /nix/store/dj9azask9bdnp3lzni7qjj68aj5g2bjg-vimplugin-persisted-.drv'.
error: 1 dependencies of derivation '/nix/store/96nhbl9jnicaxbczlxx25x4kyswf9j9j-packdir-start.drv' failed to build
error: 1 dependencies of derivation '/nix/store/1768rm7gqb0vi6arpq88s0bwy824sdbk-vim-pack-dir.drv' failed to build
error: 1 dependencies of derivation '/nix/store/jscyrrr6js0plp5fksdn73q1pn9brpdx-neovim-33e1a8c.drv' failed to build
error: 1 dependencies of derivation '/nix/store/92h0v6vm08nf1xv7i8yr456rh0cd9ww2-home-manager-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/lbnms815d89kx6qg73ncjn4l35wgqb4y-home-manager-generation.drv' failed to build
error: 1 dependencies of derivation '/nix/store/1gyxvdpwqpfkbqk0zdczd70pm7jm0qcg-user-environment.drv' failed to build
error: 1 dependencies of derivation '/nix/store/qigc6rccrnj7fc2wyxamk4ikzcmq0p4m-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/qgd1rm6dig3hq6xpgva01021j6h8mwdw-nixos-system-hostname-23.11.20230820.0b07d49.drv' failed to build
make: *** [Makefile:52: rebuild] Error 1
```

`stylua` is avail globally under nixos, but dues to its nature, it's not from /usr/bin, idk if this plugin harcode somewhere that prevent it work under nixos.

**Comment by olimorris** [comment]
What does your config look like? Can't say I've ever used Nix so can't really help

**Reply by tmpm697** [reply]
I saw two lines:
```
persisted.nvim/Makefile
28:	stylua --check lua/ tests/ -f ./stylua.toml
31:	stylua lua/ tests/ -f ./stylua.toml
```
It hasn't loaded up ok yet, so its config doesn't matter.

If it does matter, I just call setup() function with empty `require("persisted").setup({})`

**Reply by olimorris** [reply]
Does nixos call the Makefile? That file only exists to run tests in dev

**Reply by tmpm697** [reply]
Idk but from what I've tested, remove Makefile the plugin load ok, but i got this error when issue :SessionSave:
```
E5108: Error executing lua .../myNeovimPackages/start/persisted/lua/persisted/init.lua:66: attempt to concatenate field 'branch_separator' (a nil value)
stack traceback:
        .../myNeovimPackages/start/persisted/lua/persisted/init.lua:66: in function 'get_branch'
        .../myNeovimPackages/start/persisted/lua/persisted/init.lua:73: in function 'get_current'
        .../myNeovimPackages/start/persisted/lua/persisted/init.lua:171: in function 'save'
        [string ":lua"]:1: in main chunk
```
NVIM v0.10.0-dev-33e1a8c, HEAD persisted

**Reply by olimorris** [reply]
Could this be because you're passing an empty table to the setup function? It's defined in the `config.lua` file

**Reply by tmpm697** [reply]
from what i've tested, i can assure that having `Makefile` under this root project folder, nixpkgs will invoke --> this dues too some kind of assumption for testing plugin maybe? i don't quite sure but remove it make it build ok. (or might be when building, it put the folder under a sanbox that `stylua` is not avail)

this plugin appeaared in nixpkgs but recent deleted i think, you can check that it listed [here](https://github.com/NixNeovim/NixNeovimPlugins/blob/main/plugins.md) but if you pull it, it actually not found currently.

Can you help to have Makefile always return ok? or move it somewhere else in sub-folder?

**Reply by olimorris** [reply]
Just wanted to pass on [this](https://github.com/olimorris/onedarkpro.nvim/issues/217) discussion from my other plugin and in particular [this](https://github.com/olimorris/onedarkpro.nvim/issues/217#issuecomment-1813078073) comment
